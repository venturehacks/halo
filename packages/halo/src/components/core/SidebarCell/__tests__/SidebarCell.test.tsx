import { mount, render, shallow } from 'enzyme';
import React from 'react';

import { SidebarCell, SidebarCellProps } from '../index';

const basicProps: SidebarCellProps = {
  byline: 'California, Utah, Colorado, Remote',
  isSelected: true,
  title: 'Account Executive',
};

const completeProps: SidebarCellProps = {
  ...basicProps,
  header: 'Last edit: 7 days ago',
  isStarred: true,
  onClick: () => alert('onClick'),
  onStarClick: () => alert('onStarClick'),
};

describe('SidebarCell', () => {
  test('smoke', () => {
    const component = shallow(<SidebarCell {...basicProps} />);
    expect(component.text()).toEqual(
      'Account ExecutiveCalifornia, Utah, Colorado, Remote',
    );
  });

  test('complete', () => {
    const component = shallow(<SidebarCell {...completeProps} />);
    expect(component.text()).toEqual(
      'Last edit: 7 days agoAccount Executive<StarFillIcon />California, Utah, Colorado, Remote',
    );
  });

  test('arbitrary children', () => {
    const component = shallow(
      <SidebarCell {...completeProps}>Posted by: foobar</SidebarCell>,
    );
    expect(component.text()).toEqual(
      'Last edit: 7 days agoAccount Executive<StarFillIcon />California, Utah, Colorado, RemotePosted by: foobar',
    );
  });

  test('cell callback', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <SidebarCell {...completeProps} onClick={callback} />,
    );

    expect(callback).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
  });

  test('star callback', () => {
    const callback = jest.fn();
    const starCallback = jest.fn();
    const wrapper = mount(
      <SidebarCell
        {...completeProps}
        onClick={callback}
        onStarClick={starCallback}
      />,
    );

    const starButton = wrapper.find('button');

    // clicking on star should not run parent onClick
    starButton.simulate('click');
    expect(starCallback).toHaveBeenCalledTimes(1);
    expect(callback).toHaveBeenCalledTimes(0);

    // parent onClick should not trigger starCallback
    wrapper.simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
    expect(starCallback).toHaveBeenCalledTimes(1);
  });

  describe('snapshot', () => {
    test('active', () => {
      const component = render(<SidebarCell {...basicProps} isSelected />);
      expect(component).toMatchSnapshot();
    });

    test('starred', () => {
      const component = render(<SidebarCell {...basicProps} isStarred />);
      expect(component).toMatchSnapshot();
    });

    test('accessibility attributes from onClick', () => {
      const component = render(<SidebarCell {...completeProps} />);
      expect(component).toMatchSnapshot();
    });

    test('data-attribute', () => {
      const component = render(<SidebarCell {...basicProps} data-id="123" />);
      expect(component).toMatchSnapshot();
    });
  });
});
