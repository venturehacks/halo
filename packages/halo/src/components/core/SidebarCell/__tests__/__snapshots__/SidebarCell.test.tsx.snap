// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarCell snapshot accessibility attributes from onClick 1`] = `
<div
  aria-label="Account Executive"
  class="flex flex-col justify-start content-start px-6 py-4 border-b border-slate-200 hover:bg-slate-300 rounded cursor-pointer component"
  data-test="SidebarCell"
  role="menuitem"
  tabindex="0"
>
  <div
    class="flex flex-col md:flex-row md:items-center component isolateFirstChild gap-4 text-2xs text-dark-a"
  >
    Last edit: 7 days ago
  </div>
  <div
    class="flex flex-col md:flex-row md:items-center component isolateFirstChild gap-4"
  >
    <div
      class="text-sm selected max-w-80"
    >
      <div
        class="truncate"
      >
        Account Executive
      </div>
    </div>
    <button
      aria-checked="true"
      aria-label="Star 'Account Executive'"
      role="switch"
    >
      <svg
        class="haloIcon"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="m11.444 19.478-4.329 2.378c-.584.321-1.307.086-1.615-.524a1.3 1.3 0 0 1-.12-.793l.827-5.037a1.285 1.285 0 0 0-.344-1.106L2.36 10.83a1.29 1.29 0 0 1-.021-1.767c.183-.197.424-.325.684-.364l4.84-.735c.389-.06.725-.315.9-.684l2.164-4.582c.292-.619 1.009-.873 1.601-.568.236.122.427.321.543.568l2.165 4.582c.174.369.51.625.9.684l4.839.735c.654.099 1.107.733 1.012 1.416a1.27 1.27 0 0 1-.35.715l-3.5 3.567c-.283.287-.411.7-.345 1.106l.827 5.037c.112.68-.326 1.326-.977 1.443-.259.046-.526.002-.758-.126l-4.328-2.378a1.15 1.15 0 0 0-1.114 0Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
  <div
    class="text-xs text-dark-aa"
  >
    <div>
      California, Utah, Colorado, Remote
    </div>
  </div>
</div>
`;

exports[`SidebarCell snapshot active 1`] = `
<div
  aria-label="Account Executive"
  class="flex flex-col justify-start content-start px-6 py-4 border-b border-slate-200 hover:bg-slate-300 rounded cursor-pointer component"
  data-test="SidebarCell"
>
  <div
    class="text-sm selected max-w-full"
  >
    <div
      class="truncate"
    >
      Account Executive
    </div>
  </div>
  <div
    class="text-xs text-dark-aa"
  >
    <div>
      California, Utah, Colorado, Remote
    </div>
  </div>
</div>
`;

exports[`SidebarCell snapshot data-attribute 1`] = `
<div
  aria-label="Account Executive"
  class="flex flex-col justify-start content-start px-6 py-4 border-b border-slate-200 hover:bg-slate-300 rounded cursor-pointer component"
  data-id="123"
  data-test="SidebarCell"
>
  <div
    class="text-sm selected max-w-full"
  >
    <div
      class="truncate"
    >
      Account Executive
    </div>
  </div>
  <div
    class="text-xs text-dark-aa"
  >
    <div>
      California, Utah, Colorado, Remote
    </div>
  </div>
</div>
`;

exports[`SidebarCell snapshot starred 1`] = `
<div
  aria-label="Account Executive"
  class="flex flex-col justify-start content-start px-6 py-4 border-b border-slate-200 hover:bg-slate-300 rounded cursor-pointer component"
  data-test="SidebarCell"
>
  <div
    class="text-sm selected max-w-full"
  >
    <div
      class="truncate"
    >
      Account Executive
    </div>
  </div>
  <div
    class="text-xs text-dark-aa"
  >
    <div>
      California, Utah, Colorado, Remote
    </div>
  </div>
</div>
`;
